packages/fields/src/types/Relationship/views/CreateItemModal.js

```diff
@@ -1,237 +0,0 @@
-/** @jsx jsx */
```

Comment:
This file is close, but _not_ equivalent to `packages/app-admin-ui/client/components/CreateItemModal.js`:

```
git diff --no-index packages/app-admin-ui/client/components/CreateItemModal.js packages/fields/src/types/Relationship/views/CreateItemModal.js
```

```diff
diff --git a/packages/app-admin-ui/client/components/CreateItemModal.js b/packages/fields/src/types/Relationship/views/CreateItemModal.js
index 607b2390..41c2e8ce 100644
--- a/packages/app-admin-ui/client/components/CreateItemModal.js
+++ b/packages/fields/src/types/Relationship/views/CreateItemModal.js
@@ -2,23 +2,34 @@
 import { jsx } from '@emotion/core';
 import { Component, Fragment, useCallback, useMemo, Suspense } from 'react';
 import { Mutation } from 'react-apollo';
-import { useToasts } from 'react-toast-notifications';
 
 import { Button, LoadingButton } from '@arch-ui/button';
 import Drawer from '@arch-ui/drawer';
 import { arrayToObject, captureSuspensePromises, countArrays } from '@keystonejs/utils';
 import { gridSize } from '@arch-ui/theme';
 import { AutocompleteCaptor } from '@arch-ui/input';
-
-import PageLoading from './PageLoading';
-import { validateFields, toastError } from '../util';
+import { LoadingIndicator } from '@arch-ui/loading';
+
+const PageLoading = () => (
+  <div
+    css={{
+      height: 200,
+      display: 'flex',
+      alignItems: 'center',
+      justifyContent: 'center',
+      width: '100%',
+    }}
+  >
+    <LoadingIndicator size={12} />
+  </div>
+);
 
 let Render = ({ children }) => children();
 
 class CreateItemModal extends Component {
   constructor(props) {
     super(props);
-    const { list, prefillData } = props;
+    const { list, prefillData = {} } = props;
     const item = list.getInitialItemData({ prefill: prefillData });
     const validationErrors = {};
     const validationWarnings = {};
@@ -41,7 +52,6 @@ class CreateItemModal extends Component {
       list: { fields },
       createItem,
       isLoading,
-      addToast,
     } = this.props;
     if (isLoading) return;
     const { item, validationErrors, validationWarnings } = this.state;
@@ -57,7 +67,29 @@ class CreateItemModal extends Component {
     const data = arrayToObject(creatable, 'path', field => field.serialize(item));
 
     if (!countArrays(validationWarnings)) {
-      const { errors, warnings } = await validateFields(creatable, item, data);
+      const errors = {};
+      const warnings = {};
+
+      await Promise.all(
+        fields.map(({ validateInput, path }) => {
+          const addFieldValidationError = (message, data) => {
+            errors[path] = errors[path] || [];
+            errors[path].push({ message, data });
+          };
+
+          const addFieldValidationWarning = (message, data) => {
+            warnings[path] = warnings[path] || [];
+            warnings[path].push({ message, data });
+          };
+
+          return validateInput({
+            resolvedData: data,
+            originalInput: item,
+            addFieldValidationError,
+            addFieldValidationWarning,
+          });
+        })
+      );
 
       if (countArrays(errors) + countArrays(warnings) > 0) {
         this.setState(() => ({
@@ -71,14 +103,10 @@ class CreateItemModal extends Component {
 
     createItem({
       variables: { data },
-    })
-      .then(data => {
-        this.props.onCreate(data);
-        this.setState({ item: this.props.list.getInitialItemData({}) });
-      })
-      .catch(error => {
-        toastError({ addToast, options: { autoDismiss: true } }, error);
-      });
+    }).then(data => {
+      this.props.onCreate(data);
+      this.setState({ item: this.props.list.getInitialItemData({}) });
+    });
   };
   onClose = () => {
     const { isLoading } = this.props;
@@ -100,7 +128,7 @@ class CreateItemModal extends Component {
     const hasWarnings = countArrays(validationWarnings);
     const hasErrors = countArrays(validationErrors);
 
-    const cypressId = 'create-item-modal-submit-button';
+    const cypressId = 'create-relationship-item-modal-submit-button';
 
     return (
       <Drawer
@@ -144,46 +172,48 @@ class CreateItemModal extends Component {
                   .filter(({ maybeAccess }) => !!maybeAccess.create);
 
                 captureSuspensePromises(creatable.map(field => () => field.initFieldView()));
-                return creatable.map((field, i) => (
-                  <Render key={field.path}>
-                    {() => {
-                      let [Field] = field.adminMeta.readViews([field.views.Field]);
-                      let onChange = useCallback(value => {
-                        this.setState(({ item }) => ({
-                          item: {
-                            ...item,
-                            [field.path]: value,
-                          },
-                          validationErrors: {},
-                          validationWarnings: {},
-                        }));
-                      }, []);
-                      return useMemo(
-                        () => (
-                          <Field
-                            autoFocus={!i}
-                            value={item[field.path]}
-                            savedValue={item[field.path]}
-                            field={field}
-                            /* TODO: Permission query results */
-                            errors={validationErrors[field.path] || []}
-                            warnings={validationWarnings[field.path] || []}
-                            onChange={onChange}
-                            renderContext="dialog"
-                          />
-                        ),
-                        [
-                          i,
-                          item[field.path],
-                          field,
-                          onChange,
-                          validationErrors[field.path],
-                          validationWarnings[field.path],
-                        ]
-                      );
-                    }}
-                  </Render>
-                ));
+                return creatable.map((field, i) => {
+                  return (
+                    <Render key={field.path}>
+                      {() => {
+                        let [Field] = field.adminMeta.readViews([field.views.Field]);
+                        let onChange = useCallback(value => {
+                          this.setState(({ item }) => ({
+                            item: {
+                              ...item,
+                              [field.path]: value,
+                            },
+                            validationErrors: {},
+                            validationWarnings: {},
+                          }));
+                        }, []);
+                        return useMemo(
+                          () => (
+                            <Field
+                              autoFocus={!i}
+                              value={item[field.path]}
+                              savedValue={item[field.path]}
+                              field={field}
+                              /* TODO: Permission query results */
+                              errors={validationErrors[field.path] || []}
+                              warnings={validationWarnings[field.path] || []}
+                              onChange={onChange}
+                              renderContext="dialog"
+                            />
+                          ),
+                          [
+                            i,
+                            item[field.path],
+                            field,
+                            onChange,
+                            validationErrors[field.path],
+                            validationWarnings[field.path],
+                          ]
+                        );
+                      }}
+                    </Render>
+                  );
+                });
               }}
             </Render>
           </Suspense>
@@ -193,19 +223,15 @@ class CreateItemModal extends Component {
   }
 }
 
-export default function CreateItemModalWithMutation(props) {
-  const { list } = props;
-  const { addToast } = useToasts();
-  return (
-    <Mutation mutation={list.createMutation}>
-      {(createItem, { loading }) => (
-        <CreateItemModal
-          createItem={createItem}
-          isLoading={loading}
-          addToast={addToast}
-          {...props}
-        />
-      )}
-    </Mutation>
-  );
+export default class CreateItemModalWithMutation extends Component {
+  render() {
+    const { list } = this.props;
+    return (
+      <Mutation mutation={list.createMutation}>
+        {(createItem, { loading }) => (
+          <CreateItemModal createItem={createItem} isLoading={loading} {...this.props} />
+        )}
+      </Mutation>
+    );
+  }
 }
```

Reply:
these changes are duplicated to Relationship version inline type instead of copying other imports.  I have merge improvements in both files to one.